message_tagging_service/tagging_service.py

```diff
@@ -274,6 +275,42 @@ def match(self, modulemd):
             return RuleMatch(False)
 
 
+def login_koji(session, config):
+    """Log into Koji
+
+    By default, client is authenticated via Kerberos, which is set in Koji
+    configuration. However, it could be changed to autheticate client via SSL
+    by setting ``koji_cert`` in MTS configuration.
+
+    Please note that, only both ``keytab`` and ``principal`` are set in
+    configuration, the session could be logged in with a specific keytab.
+
+    :param session: an object of ClientSession to login.
+    :type session: koji.ClientSession
+    :param dict config: a mapping containing koji config. Generally, it is the
+        return value from ``koji.read_config``.
+    """
+    # Loaded koji config is a simple dict without nested dict. dict.copy just
+    # works well.
+    cfg = config.copy()
+    use_ssl = conf.koji_cert is not None
+    if use_ssl:
```

Comment:
Minor: I wonder why this is needed. Can't it just be fully driven by the available koji configuration?

Reply:
It could be. I just thought we can easily control MTS by the configs inside MTS itself. :)